### PR TITLE
CEDS 1368 Qualifier code not present in payload

### PR DESCRIPTION
**Pull request by benjaminparker**
Missing Qualifier Code is now in the model  and swapping goodsLocation.id and goodsLocation.name as they were interchanged (secondary bug discovered)

### CODE REVIEW

File: test/services/mapping/goodsshipment/consignment/GoodsLocationBuilderSpec.scala [hunk 1]
```diff
@@ -15,31 +15,31 @@
...
-            "CacheID",
-            Map(
-              GoodsLocation.formId ->
-                GoodsLocationSpec.correctGoodsLocationJSON
-            )
-          )
+          CacheMap("CacheID", Map(GoodsLocation.formId -> correctGoodsLocationJSON))
+        
         val goodsLocation = GoodsLocationBuilder.build
-        goodsLocation.getID.getValue should be("9GB1234567ABCDEF")
```
Comment (mattclark-zerogravit) [inline test/services/mapping/goodsshipment/consignment/GoodsLocationBuilderSpec.scala]: Again we agreed as a team we are happy with the should be() approach and this just introduces inconsistency in our tests. Please leave as should be() unless agreed with team we are all moving to using shouldBe

File: app/services/mapping/goodsshipment/consignment/GoodsLocationBuilder.scala [hunk 2]
```diff
@@ -56,9 +57,9 @@ object GoodsLocationBuilder {
     }
 
     goods.identificationOfLocation.foreach { value =>
-      val name = new GoodsLocationNameTextType()
-      name.setValue(value)
-      goodsLocation.setName(name)
+      val id = new GoodsLocationIdentificationIDType()
```
Comment (PauloPortugal) [inline app/services/mapping/goodsshipment/consignment/GoodsLocationBuilder.scala]: :+1:

File: app/services/mapping/goodsshipment/consignment/GoodsLocationBuilder.scala [hunk 3]
```diff
@@ -43,10 +43,11 @@ object GoodsLocationBuilder {
   private def buildEoriOrAddress(goods: GoodsLocation) = {
     val goodsLocation = new Consignment.GoodsLocation()
 
-    goods.additionalIdentifier.foreach { value =>
-      val id = new GoodsLocationIdentificationIDType()
-      id.setValue(value)
-      goodsLocation.setID(id)
+    goods.additionalQualifier.foreach { value =>
```
Comment (PauloPortugal) [inline app/services/mapping/goodsshipment/consignment/GoodsLocationBuilder.scala]: :+1:

File: test/resources/wco_dec_metadata.xml [hunk 4]
```diff
@@ -75,10 +75,11 @@
                     <ns3:IdentificationTypeCode>10</ns3:IdentificationTypeCode>
                 </ns3:DepartureTransportMeans>
                 <ns3:GoodsLocation>
-                    <ns3:Name>LOC</ns3:Name>
-                    <ns3:ID>9GB1234567ABCDEF</ns3:ID>
+                    <ns3:Name>9GB1234567ABCDEF</ns3:Name>
```
Comment (PauloPortugal) [inline test/resources/wco_dec_metadata.xml]: :+1:

File: test/services/mapping/declaration/DeclarationBuilderSpec.scala [hunk 5]
```diff
@@ -25,25 +25,25 @@ class DeclarationBuilderSpec extends WordSpec with Matchers {
   "DeclarationBuilder" should {
     "correctly map a Supplementary declaration to the WCO-DEC Declaration instance" in {
       val declaration =
-        DeclarationBuilder.build(SupplementaryDeclarationDataSpec.cacheMapAllRecords, supplementaryChoice)
+        DeclarationBuilder.build(SupplementaryDeclarationTestData.cacheMapAllRecords, supplementaryChoice)
 
       declaration.getAgent.getID.getValue should be("9GB1234567ABCDEF")
-      declaration.getAgent.getName should be(null)
```
[the remark this answers is not in the document]
Comment (benjaminparker) [reply test/services/mapping/declaration/DeclarationBuilderSpec.scala]: I think this needs discussion as keeping extra boiler-plate code is making tests harder to read.  Will change all back to () style now.

**Pull request merged**